enhancements/installer/agent-based-installation-in-hive.md

```diff
@@ -0,0 +1,647 @@
+that ClusterDeployment will need to consume Agents from a different namespace.
+But the Infra Owner may not want them to have the ability to consume any Agent
+from any namespace.
+
+To designate that a ClusterDeployment is authorized to consume Agents from a
+particular InstallEnv, a reference to that ClusterDeployment must be present
+in an array on the InstallEnv. The reference proves that someone with write
+access to the InstallEnv approves of associating it with the cluster.
+
+The restriction will be enforced by the agent-related controllers at the time
+of matching Agents to a particular ClusterDeployment. The controllers will
+still have access to all namespaces.
+
+InstallEnv and Agent resources are both namespaced. Agents will be created in
+the same namespace as the InstallEnv from which they were created.
+
+In the 4.8 timeframe, because "late binding" does not exist and thus each
+InstallEnv must be associated to a single cluster, this array of references
+will be required to have exactly one member. Once late binding is supported,
+that restriction will be removed. ("late binding" refers to the ability to
+create a discovery ISO and boot Agents with it prior to associating them with a
+cluster.)
+
+#### Create Cluster
+
+This scenario takes place on a hub cluster where hive and possibly RHACM are
+present. This scenario does not include centralized machine management.
+
+This scenario is an end-to-end flow that enables the user to specify everything
+up-front and then have an automated process create a cluster. Because the
+user's primary frame of reference is hardware-oriented, this flow enables them
+to specify Node attributes such as "role" with their BareMetalHost definition.
+Specifying those attributes together is particularly useful in a gitops
+approach.
+
+1. Infra Owner creates an InstallEnv resource. It can include fields such as egress proxy, NTP server, SSH public key, ... In particular it includes an Agent label selector, and a separate field of labels that will be automatically applied to Agents.
+1. Infra Owner creates BareMetalHost resources with corresponding BMC credentials. They must be labeled so that they match the selctor on the InstallEnv and must be in the same namespace as the InstallEnv.
+1. A new controller, the Baremetal Agent Controller, sees the matching BareMetalHosts and boots them using the discovery ISO URL found in the InstallEnv's status.
+1. The Agent starts up on each host and reports back to the assisted service, which creates an Agent resource in the same namespace as the InstallEnv. The Agent is automatically labeled with the labels that were specified in the InstallEnv's Spec.
+1. The Baremetal Agent Controller sets the Agent's Role field in its spec to "master" or "worker" if a corresponding label was present on its BareMetalHost.
+1. The Agent is automatically marked as Approved via a field in its Spec based on being recognized as running on the known BareMetalHost.
```

Comment:
What does this, the Baremetal Agent Controller? How does it match the Agent to the host?

Reply:
Yes.  The idea was that the BMAC would match a MAC address in the Agent's inventory to the MAC address on the BMH.